aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -102,8 +103,17 @@ Completable hydrate() {
 
     private Completable createHydrationTask(
             ModelWithMetadataComparator modelWithMetadataComparator, Class<? extends Model> modelClass) {
+        // Default this to BASE since if there is no LastSyncMetadata record for this
+        // model, a BASE sync will be performed.
+        AtomicReference<SyncType> syncType = new AtomicReference<>(SyncType.BASE);
+
         return syncTimeRegistry.lookupLastSyncTime(modelClass)
             .map(this::filterOutOldSyncTimes)
+            .map(lastSyncTime -> {
+                long syncIntervalMs = dataStoreConfigurationProvider.getConfiguration().getSyncIntervalMs();
+                syncType.set(SyncType.fromSyncTimeAndThreshold(lastSyncTime, syncIntervalMs));
+                return lastSyncTime;
```

Comment:
Seems like we should just keep the return type from this `.map(...)` as `LastSyncMetadata`, here. If you do that, you won't need the `AtomicReference`. And, you can access the component data in the `syncModel(...)` call, below, by pulling out `getModelClass()` and `getLastSyncTime()`.

Reply:
I refactored it a bit so we don't need that AtomicReference. Let me know if it makes sense. I did keep the return type of the map as `SyncTime` because we don't have the storage adapter available to get the `LastSyncMetadata` record.